sstables/mx/writer.cc

```diff
@@ -844,6 +852,12 @@ void writer::add_pi_block() {
 
     write_pi_block(block);
     ++_pi_write_m.promoted_index_size;
+
+    // auto-scale?
+    if (_pi_write_m.auto_scale_threshold && _pi_write_m.blocks.size() >= _pi_write_m.auto_scale_threshold) {
```

Comment:
Better to use max for the threshold to avoid special values (0).

Reply:
Meaning tell the user to set to <size_t>::max() (or to an arbitrary high value) to effectively disable?